### PR TITLE
Add universal Cell server.  

### PR DESCRIPTION
**Pull request by vEpiphyte**
* Add a universal Cell server.
* Fix bug with synapse.lib.reflect.getShareInfo()

### CODE REVIEW

File: synapse/servers/cell.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+def parse(argv):
+    cctor = os.getenv('SYN_UNIV_CELL', 'synapse.lib.cell.Cell')
+    https = os.getenv('SYN_UNIV_HTTPS', '4443')
+    telep = os.getenv('SYN_UNIV_TELEPATH', 'tcp://0.0.0.0:27492/')
+    telen = os.getenv('SYN_UNIV_NAME', None)
+
+    pars = argparse.ArgumentParser(prog='synapse.servers.cell',
+                                   description='A universal Synapse Cell loader.')
+    pars.add_argument('--cell', default=cctor, dest='ctor', help='Python class path to use to load the Cell.')
```
Comment (jnwatson) [inline synapse/servers/cell.py]: default -> metavar?

File: synapse/servers/cell.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    pars.add_argument('--telepath', default=telep, help='The telepath URL to listen on.')
+    pars.add_argument('--https', default=https, dest='port', type=int, help='The port to bind for the HTTPS/REST API.')
+    pars.add_argument('--name', default=telen, help='The (optional) additional name to share the Cell as.')
+    pars.add_argument('celldir', help='The directory for the cortex to use for storage.')
```
Comment (vEpiphyte) [inline synapse/servers/cell.py]: cortex -> cell

File: synapse/servers/cell.py (hunk 1 above)
Comment (vEpiphyte) [reply synapse/servers/cell.py]: just removed the dest argument

File: synapse/servers/cell.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+def parse(argv):
+    cctor = os.getenv('SYN_UNIV_CELL', 'synapse.lib.cell.Cell')
+    https = os.getenv('SYN_UNIV_HTTPS', '4443')
+    telep = os.getenv('SYN_UNIV_TELEPATH', 'tcp://0.0.0.0:27492/')
+    telen = os.getenv('SYN_UNIV_NAME', None)
+
+    pars = argparse.ArgumentParser(prog='synapse.servers.cell',
+                                   description='A universal Synapse Cell loader.')
+    pars.add_argument('--cell', default=cctor, help='Python class path to use to load the Cell.')
```
Comment (invisig0th) [inline synapse/servers/cell.py]: This argument should be mandatory i think.  Doesn't seem to make sense to execute without it.

File: synapse/servers/cortex.py [hunk 4]
```diff
@@ -33,27 +33,17 @@ def parse(argv):
...
-        outp.printf('...cortex API (telepath): %s' % (opts.telepath,))
-        await core.dmon.listen(opts.telepath)
-
-        outp.printf('...cortex API (https): %s' % (opts.port,))
-        await core.addHttpsPort(opts.port)
-
-        if opts.name:
-            outp.printf(f'...cortex additional share name: {opts.name}')
-            core.dmon.share(opts.name, core)
+    core = await s_s_cell.getCell(outp,
```
Comment (invisig0th) [inline synapse/servers/cortex.py]: I dont want to funnel cell constructors through a central factory function.  It makes things messy when some cells want custom options.  I dont really see any advantage to having this abstraction in the case for built-in cells :)

File: synapse/tests/test_servers_univ.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            # Start a cortex with the universal loader
+            async with await s_s_univ.main(argu, outp=outp) as core:
+
+                async with await s_telepath.openurl(f'cell://{dirn}') as proxy:
+                    podes = await s_t_utils.alist(proxy.eval(f'[ou:org={guid}]'))
+                    self.eq('cortex', await proxy.getCellType())
+
+                self.true(core.dmon.shared.get('telecore') is core)
+
+            # And data persists... and can be seen with the regular synapse server
```
Comment (invisig0th) [inline synapse/tests/test_servers_univ.py]: as always... excellent tests!

File: synapse/servers/cortex.py (hunk 4 above)
Comment (vEpiphyte) [reply synapse/servers/cortex.py]: backed these out

File: synapse/servers/cell.py (hunk 3 above)
Comment (vEpiphyte) [reply synapse/servers/cell.py]: So it starts with a default value of `synapse.lib.cell.Cell` and is controlled via a envar.

Comment (vEpiphyte) [reply synapse/servers/cell.py]: made the ctor required

**Pull request merged**